.github/scripts/pre-commit.sh

```diff
@@ -61,7 +61,7 @@ runShellCheckForCommitFiles() {
       filename=$(basename -- "$file")
       extension="${filename##*.}"
       # check for a hashbang or a .sh extension to determine if this is a shell script.
-      if [ "$firstTwo" == "#!" ] || [ "$extension" == "sh" ]; then
```

Comment:
Current naive check for existence of `#!` captures contributions to e.g. Jenkinsfile resources which can't be helped by shellcheck...